raft/server.cc

```diff
@@ -283,8 +283,22 @@ server_impl::server_impl(server_id uuid, std::unique_ptr<rpc> rpc,
                     _persistence(std::move(persistence)), _failure_detector(failure_detector),
                     _id(uuid), _config(config) {
     set_rpc_server(_rpc.get());
-    if (_config.snapshot_threshold > _config.max_log_size) {
-        throw config_error("snapshot_threshold has to be smaller than max_log_size");
+    if (_config.snapshot_threshold_log_size > _config.max_log_size) {
+        throw config_error(fmt::format("[{}] snapshot_threshold_log_size ({}) must not be greater than max_log_size ({})",
+            _id, _config.snapshot_threshold_log_size, _config.max_log_size));
+    }
+    if (_config.snapshot_trailing_size > _config.snapshot_threshold_log_size) {
+        throw config_error(fmt::format("[{}] snapshot_trailing_size ({}) must not be greater than snapshot_threshold_log_size ({})",
+                                       _id, _config.snapshot_trailing_size, _config.snapshot_threshold_log_size));
+    }
+    if (_config.max_command_memory_usage() > _config.max_log_size - _config.snapshot_trailing_size) {
+        throw config_error(fmt::format(
+            "[{}] max_command_size + log_entry size ({} + {} = {}) must not be greater than "
```

Comment:
then may be better to write sizeof(log_entry) since log_entry size is ambiguous. Though we need to think if we can avoid sizeof(log_entry) dependency from the API. It makes its size not only API but also ABI which is much stronger.